src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -352,75 +456,6 @@ private String getGitBranchName(Supplier<String> gitCommand) {
-
-        private final BuildScanExtension buildScan;
-        private final Map<String, String> customValueLinks;
-
-        private CustomValueSearchLinker(BuildScanExtension buildScan) {
-            this.buildScan = buildScan;
-            this.customValueLinks = new LinkedHashMap<>();
-        }
-
-        private static CustomValueSearchLinker registerWith(BuildScanExtension buildScan) {
-            CustomValueSearchLinker customValueSearchLinker = new CustomValueSearchLinker(buildScan);
-            buildScan.buildFinished(customValueSearchLinker);
-            return customValueSearchLinker;
-        }
-
-        private void addCustomValueAndSearchLink(String name, String value) {
-            buildScan.value(name, value);
-            registerLink(name, name, value);
-        }
-
-        public void addCustomValueAndSearchLink(String linkLabel, String name, String value) {
-            buildScan.value(name, value);
-            registerLink(linkLabel, name, value);
-        }
-
-        private void registerLink(String linkLabel, Map<String, String> values) {
-            // the parameters for a link querying multiple custom values look like:
-            // search.names=name1,name2&search.values=value1,value2
-            // this reduction groups all names and all values together in order to properly generate the query
-            values.entrySet().stream()
-                .sorted(Map.Entry.comparingByKey()) // results in a deterministic order of link parameters
-                .reduce((a, b) -> new AbstractMap.SimpleEntry<>(a.getKey() + "," + b.getKey(), a.getValue() + "," + b.getValue()))
-                .ifPresent(x -> registerLink(linkLabel, x.getKey(), x.getValue()));
-        }
-
-        private synchronized void registerLink(String linkLabel, String name, String value) {
-            String searchParams = "search.names=" + urlEncode(name) + "&search.values=" + urlEncode(value);
-            customValueLinks.put(linkLabel, searchParams);
-        }
-
-        @Override
-        public synchronized void execute(BuildResult buildResult) {
-            String server = getServer(buildScan);
-            if (server != null) {
-                customValueLinks.forEach((linkLabel, searchParams) -> {
-                    String url = appendIfMissing(server, "/") + "scans?" + searchParams + "#selection.buildScanB=" + urlEncode("{SCAN_ID}");
-                    buildScan.link(linkLabel + " build scans", url);
-                });
-            }
-        }
-
-        private static String getServer(BuildScanExtension buildScan) {
-            try {
-                buildScan.getClass().getMethod("getServer");
-                return buildScan.getServer();
-            } catch (NoSuchMethodException e) {
-                // not available in Gradle 4.x / Build Scan plugin 1.x
-                return null;
-            }
-        }
-
-    }
-
     private void captureTestParallelization() {
```

Comment:
can we move this to the bottom of the file like it used to be to lessen the diff? or is there a reason this grouping makes more logical sense?

Reply:
I grouped all the static methods together at the end of the file, there was originally only 1, we could indeed move the `captureTestParallelization`  method but I find the actual setup better.